darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterStrategyFactoryImpl.java

```diff
@@ -63,13 +66,15 @@
   private final Random _random;
   private final LoadBalancerClusterListener _clusterListener;
   private final DarkClusterVerifierManager _verifierManager;
+  private final ConstantQpsRateLimiter _rateLimiter;
 
   public DarkClusterStrategyFactoryImpl(@Nonnull Facilities facilities,
                                         @Nonnull String sourceClusterName,
                                         @Nonnull DarkClusterDispatcher darkClusterDispatcher,
                                         @Nonnull Notifier notifier,
                                         @Nonnull Random random,
-                                        @Nonnull DarkClusterVerifierManager verifierManager)
+                                        @Nonnull DarkClusterVerifierManager verifierManager,
+                                        @Nonnull ConstantQpsRateLimiter rateLimiter)
```

Comment:
It's better to make the constructor backward compatible, so other external MP's based on this library won't break. One option is to keep the old constructor, and call `this(facilities, ..., verifierManager, null)`. And when rateLimiter is `null`, we don't need to support constant QPS strategy